Orange/data/variable.py

```diff
@@ -1019,3 +1004,19 @@ def to_val(self, s):
             return self.parse(s)
         else:
             return super().to_val(s)
+
+
+class _TypeIndicatorDescriptor:
+    def __init__(self, atype):
+        self.check_type = lambda instance: isinstance(instance, atype)
```

Comment:
Make a named function, and a __private one.

Reply:
PEP 8 recommends `def f(x): return 2 * x` over `f = lambda x: 2 * x`. This doesn't really apply here since I can't assign the proper name to the function (`def self.check_type(instance): ...` ?! :)). But I agree that it's clearer with a named function due to lambda's awful syntax.

Reply:
Lambdas are perfectly clean. My only concern was they are not picklable.

Reply:
Turning it into a (local) function doesn't help. But this is not a problem here, since this `self` is not an instance of `Variable` but of `_TypeIndicatorDescriptor`, which is a part of the `Variable` class and is created when importing the module. 

That is, you indeed can't pickle `_TypeIndicatorDescriptor` (precisely because of `check_type`),

```
>>> pickle.dumps(DiscreteVariable.is_discrete)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: Can't pickle local object '_TypeIndicatorDescriptor.__init__.<locals>.__check_type'
```

but `_TypeIndicatorDescriptor` is never pickled anyway. `Variable` class is still picklable, and hence are its instances.

```
>>> pickle.dumps(DiscreteVariable)
b'\x80\x03cOrange.data.variable\nDiscreteVariable\nq\x00.'
```